### PR TITLE
Fix sparkbars are not aligned

### PR DESCRIPTION
**Pull request by nicelulu**
Changelog category (leave one):

- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix sparkbars are not aligned, see: https://github.com/ClickHouse/ClickHouse/issues/26175#issuecomment-960353867,  [comment](https://github.com/ClickHouse/ClickHouse/issues/26175#issuecomment-961155065).

### CODE REVIEW

File: src/AggregateFunctions/AggregateFunctionSparkbar.h [hunk 1]
```diff
@@ -100,8 +89,9 @@ class AggregateFunctionSparkbar final
 
 private:
     size_t width;
-    X min_x;
-    X max_x;
+    mutable X min_x;
+    mutable X max_x;
+    bool specified_min_max_x;
```
Comment (vdimir) [inline src/AggregateFunctions/AggregateFunctionSparkbar.h:94]: Looks like it's better to store it in aggregate function data, but not in function itself. Is there any reason to put it here?

Comment (nicelulu) [reply src/AggregateFunctions/AggregateFunctionSparkbar.h:94]: If ```min_x```, ```max_x``` are not specified, we need to generate a global ```min_x```, ```max_x``` based on the data, and the later ```render```  method will rely on them to do sparkbar alignment.

**Comment by alexey-milovidov** [comment]
If min_x and max_x are not specified, it's ok to have unaligned sparkbars, so every sparkbar will have its own boundaries.

File: src/AggregateFunctions/AggregateFunctionSparkbar.h (hunk 1 above)
Comment (vdimir) [reply src/AggregateFunctions/AggregateFunctionSparkbar.h:94]: Why we can't do it with `AggregateFunctionSparkbarData`? Storing data in function itself ma cause issues if query would be distributed.

Comment (nicelulu) [reply src/AggregateFunctions/AggregateFunctionSparkbar.h]: You're right.

**Review by vdimir — APPROVED** [review]
lgtm

**Pull request merged**